src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -183,19 +188,38 @@ private void configureClient(DeviceConfiguration deviceConfiguration) {
             configureHttpClient(deviceConfiguration);
         }
         logger.atDebug().log(CONFIGURING_GGV2_INFO_MESSAGE);
-        GreengrassV2DataClientBuilder clientBuilder = GreengrassV2DataClient.builder()
+        String greengrassServiceEndpoint = ClientConfigurationUtils
+                .getGreengrassServiceEndpoint(deviceConfiguration);
+
+        if (Coerce.toBoolean(deviceConfiguration.getFipsMode())) {
+            GreengrassV2DataEndpointProvider endpointProvider = new GreengrassV2DataEndpointProvider() {
+
+                @Override
+                public CompletableFuture<Endpoint> resolveEndpoint(GreengrassV2DataEndpointParams endpointParams) {
+                    return CompletableFuture.supplyAsync(() -> Endpoint.builder()
+                            .url(URI.create(greengrassServiceEndpoint))
+                            .build());
+                }
+            };
+            clientBuilder = GreengrassV2DataClient.builder()
+                // Use an empty credential provider because our requests don't need SigV4
+                // signing, as they are going through IoT Core instead
```

Comment:
Why does fips differ from regular request?

Reply:
where do you set fips mode? is that already implemented in nucleus?

Reply:
Yes, we can only override the ggdataClient endpoint to iotData fips endpoint by applying the `endpointProvider`, the `endpointOverride` function doesn't have a higher priority than the built-in function `useFips` when creating aws sdk client. 

Yes, the fips mode set up is implemented https://github.com/aws-greengrass/aws-greengrass-nucleus/commit/f188e273f27dd78ebbba4b80551a656d863b1f3d